test/component/QueueSpec.scala

```diff
@@ -56,4 +64,37 @@ class QueueSpec extends FeatureSpec
       status(deleteResult) shouldBe NO_CONTENT
     }
   }
+
+  feature("Post, pull and re-pull a message from the queue") {
+    info("As a 3rd Party system")
+    info("I want to successfully persist a notification")
+    info("So that I can pull it when needed")
+    info("And pull it again when needed")
+
+    scenario("3rd party system gets a message previously queued") {
+      Given("a message has already been queued")
+      val clientId = "aaaa"
+      val xmlBody = <xml><node>Stuff</node></xml>
+      val queueResponse = await(route(app = app, FakeRequest(POST, "/queue", Headers("x-client-id" -> clientId, "content-type" -> "application/xml"), AnyContentAsEmpty).withXmlBody(xmlBody)).value)
+      val location = queueResponse.header.headers("Location")
+      val notificationId = location.substring(location.length() - 36)
+
+      When("you make a GET based on the location header")
+      val unpulledResult = route(app, FakeRequest(GET, s"/notifications/unpulled/$notificationId", Headers("x-client-id" -> clientId), AnyContentAsEmpty)).value
+
+      Then("you will receive a 200 response")
+      status(unpulledResult) shouldBe OK
+      And("the message will be the same")
+      contentAsString(unpulledResult) shouldBe xmlBody.toString()
```

Comment:
do we need a `Thread.sleep`? if so I think there should be a comment here with a TODO

Reply:
The update to mongo happens after the `pulled` request. Not ideal. Comment and TODO added.